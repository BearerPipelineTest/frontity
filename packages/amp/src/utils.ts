import { Element } from "@frontity/html2react/types";
import { warn } from "frontity";

const httpRegexp = /^http:\/\//;

/**
 * Replace "http://" with "https://" in the src property of an HTML element.
 *
 * @param node - An HTML element.
 *
 * @returns The same element with http:// substituted with https://.
 */
export const httpToHttps = (node: Element) => {
  // AMP requires that the file is loaded over HTTPS
  if (node.props?.src?.match(httpRegexp)) {
    node.props.src = node.props.src.replace(httpRegexp, "https://");
    warn(
      `<${
        typeof node.component === "function"
          ? node.component.displayName || node.component.name
          : node.component
      }> element with src of ${
        node.props.src
      } was found but AMP requires resources to be loaded over HTTPS.\n
Frontity will update the src attribute to point to the HTTPS version but you need to ensure that the asset is available over HTTPS.`
    );
    return node;
  }
  return node;
};
