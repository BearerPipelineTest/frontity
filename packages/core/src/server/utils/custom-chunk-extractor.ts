import { ChunkExtractor } from "@loadable/server";

/**
 * Custom ChunkExtractor class to override `preload` with `modulepreload` based
 * on asset filename.
 */
export class CustomChunkExtractor extends ChunkExtractor {
  /**
   * The `getPreAssets` method.
   *
   * @returns A list of assets.
   */
  getPreAssets() {
    // eslint-disable-next-line @typescript-eslint/ban-types
    const assets = (ChunkExtractor as Function).prototype.getPreAssets.call(
      this
    );
    return assets.map((asset) => {
      if (asset.filename.indexOf(".module.") !== -1) {
        asset.linkType = "modulepreload";
      }

      return asset;
    });
  }
}
