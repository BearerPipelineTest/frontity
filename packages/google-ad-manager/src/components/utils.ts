/**
 * Turn prop names from camelCase into kebab-case, prefixing `data-`. Also,
 * filter out those props with value `undefined`.
 *
 * @param props - Component props.
 * @returns Filtered props.
 */
export const getDataProps = <Input>(props: Input): any =>
  Object.entries(props)
    .filter(([, value]) => value !== undefined)
    .map(([key, value]) => [toKebabCase(key, { prefix: "data-" }), value])
    .reduce((output, [key, value]) => ((output[key] = value), output), {});

/**
 * Convert string from camelCase to kebab-case.
 *
 * @param camelCase - String in camelCase format.
 * @param options - Options object.
 * @returns String in kebab-case, with the specified prefix.
 */
export const toKebabCase = (camelCase: string, { prefix = "" } = {}) =>
  prefix + camelCase.replace(/[A-Z]/g, (char) => `-${char.toLowerCase()}`);
