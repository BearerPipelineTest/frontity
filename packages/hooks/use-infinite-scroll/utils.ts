import memoize from "ramda/src/memoizeWith";
import { IntersectionOptions, WrapperGenerator } from "./types";

/**
 * Generate a memoize Wrapper.
 *
 * @param wrapperGenerator - The function that generates the wrapper and needs
 * to be memoized.
 *
 * @returns A memoized Wrapper.
 */
export const generateMemoizedWrapper = (wrapperGenerator: WrapperGenerator) =>
  memoize(
    (link) => link,
    (
      link: string,
      options: {
        /**
         * The intersection observer options for fetching.
         */
        fetchInViewOptions: IntersectionOptions;

        /**
         * The intersection observer options for routing.
         */
        routeInViewOptions: IntersectionOptions;
      }
    ) =>
      wrapperGenerator({
        link,
        fetchInViewOptions: options.fetchInViewOptions,
        routeInViewOptions: options.routeInViewOptions,
      })
  );
