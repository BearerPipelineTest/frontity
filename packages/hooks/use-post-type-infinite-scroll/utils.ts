import { isArchive, isRedirection } from "@frontity/source";
import Source from "@frontity/source/types";
import { State } from "frontity/types";

/**
 * Arguments passed to {@link getLinksFromPages}.
 */
type GetLinksFromPagesParams = {
  /**
   * List of links pointing to archive pages.
   *
   * @example ["/blog/", "/blog/page/2/", "/blog/page/3/"]
   */
  pages: string[];

  /**
   * The first link showed by the infinite scroll hook. The link is removed from
   * the second and subsequent pages if it's not included in the first one, in
   * order to avoid duplicated links.
   *
   * @example "/blog/first-post/"
   */
  firstLink: string;

  /**
   * The `state.source.get` state from Source.
   */
  sourceGet: State<Source>["source"]["get"];
};

/**
 * Get all the item links from the given pages, removing redirections.
 *
 * @param params - Object of type {@link GetLinksFromPagesParams}.
 * @returns List of item links.
 */
export const getLinksFromPages = ({
  pages,
  firstLink,
  sourceGet,
}: GetLinksFromPagesParams): string[] => {
  // Get the data object of all ready pages.
  const pagesData = pages
    .map((link) => sourceGet(link))
    .filter((data) => isArchive(data) && data.isReady);

  // Get all the post links from the pages.
  const rawLinks = pagesData.reduce<string[]>((allLinks, data, index) => {
    if (isArchive(data)) {
      let dataLinks = data.items.map(({ link }) => link);
      // Filter out `firstLink` if the archive page is not the first one.
      if (index > 0) dataLinks = dataLinks.filter((link) => link !== firstLink);
      allLinks = allLinks.concat(dataLinks);
    }
    return allLinks;
  }, []);

  // Remove links that point to redirections.
  return rawLinks
    .map((link) => sourceGet(link))
    .filter((data) => !isRedirection(data))
    .map(({ link }) => link);
};
