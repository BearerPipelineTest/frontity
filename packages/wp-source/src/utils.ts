import { RedirectionData } from "@frontity/source/types";
import { fetch } from "frontity";

/**
 * Return object of the {@link fetchRedirection} function.
 */
export interface FetchDirectionReturn {
  /**
   * The HTTP status.
   */
  status: 301 | 302 | 307 | 308;

  /**
   * The new link.
   */
  location: string;

  /**
   * Wheter this is a redirection or not.
   */
  isRedirection: boolean;
}

/**
 * A helper for fetching the 30x redirections.
 * Used to abstract the differences in fetching the redirections
 * between the client and the server.
 *
 * @param link - The URL from which the redirection should be fetched.
 * @param platform - Should be either "client" or "server".
 *
 * @returns An object that contains information about the redirection, defined
 * in {@link FetchDirectionReturn}, or nothing if there is no redirection.
 */
export const fetchRedirection = async (
  link: string,
  platform: "client" | "server"
): Promise<FetchDirectionReturn> => {
  if (platform === "server") {
    // On the server we have to fetch with `redirect: manual` so that we can
    // check the status code of the redirection.
    const response = await fetch(link, {
      method: "HEAD",
      redirect: "manual",
    });

    // On the server we check the status of the response.
    if (
      response.status === 301 ||
      response.status === 302 ||
      response.status === 307 ||
      response.status === 308
    )
      return {
        status: response.status,
        location: response.headers.get("location"),
        isRedirection: true,
      };
  } else {
    // On the client it's not possible to get the actual status code of the
    // redirection if you use the `redirect: manual` option because of
    // https://fetch.spec.whatwg.org/#atomic-http-redirect-handling The
    // `redirect: manual` option on the client returns an "opaque response"
    // object which always has the status of 0.
    const response = await fetch(link, {
      method: "HEAD",
    });

    // On the client we check the property `redirected`.
    if (response.redirected)
      return {
        status: 301, // Default value on the client.
        location: response.url,
        isRedirection: true,
      };
  }
};

/**
 * Check if it is an eager redirection, which means a redirection that needs
 * to be checked before we fetch the backend for the data.
 *
 * @param redirections - The redirections setting of
 * `state.source.redirections`.
 * @param link - The link to match the redirection with.
 *
 * @returns Whether or not it should try to fetch a redirection.
 */
export const isEagerRedirection = (
  redirections: string | string[],
  link: string
): boolean => {
  if (redirections === "all") {
    // It could simply all, which means that it always has to check for a
    // redirection.
    return true;
  } else if (Array.isArray(redirections)) {
    // It could be an array of RegExps and one of them matches the link.
    const patterns = redirections
      .filter((r) => r.startsWith("RegExp:"))
      .map((r) => r.replace(/^RegExp:/, ""));
    return patterns.some((r) => link.match(r));
  } else if (redirections?.startsWith("RegExp:")) {
    // Or it could be a single RegExp that matches the link.
    if (link.match(redirections.replace(/^RegExp:/, ""))) return true;
  }
};

/**
 * Check if it is a 404 redirection, which means a redirection that needs
 * to be checked after we have catched a 404 error from the backend.
 *
 * @param redirections - The redirections setting of
 * `state.source.redirections`.
 *
 * @returns Whether or not it should try to fetch a redirection.
 */
export const is404Redirection = (redirections: string | string[]): boolean => {
  return (
    redirections === "404" ||
    (Array.isArray(redirections) && redirections.includes("404"))
  );
};

/**
 * Prepare the redirection data to populate a data object that is a redirection.
 *
 * @param location - The final url of the redirection.
 * @param status - The status code of the redirection.
 * @param sourceUrl - The url of the backend, usually defined in
 * `state.source.url`.
 * @param frontityUrl - The url of the backend, usually defined in
 * `state.frontity.url`.
 *
 * @returns A data object to populate the state, defined in {@link
 * RedirectionData}.
 */
export const getRedirectionData = (
  location: string,
  status: 301 | 302 | 307 | 308,
  sourceUrl: string,
  frontityUrl: string
): Partial<RedirectionData> => ({
  isExternal:
    new URL(location).host !== new URL(sourceUrl).host &&
    new URL(location).host !== new URL(frontityUrl).host,
  location,
  redirectionStatus: status,
  [`is${status}`]: true,
  isRedirection: true,
  isReady: true,
  isFetching: false,
});
