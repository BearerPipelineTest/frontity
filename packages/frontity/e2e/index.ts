/* eslint-disable no-irregular-whitespace */
import execa from "execa";
import { readFile } from "fs-extra";
import { resolve as resolvePath } from "path";
import { testContainer } from "./utils";

// We need to set a high timeout because building the docker container and
// running `frontity create` takes a long time.
jest.setTimeout(180000);

beforeAll(async () => {
  // Remove the built output
  await execa.command("rm -rf build", { stdio: "inherit" });

  // Compile the TS source to JS
  await execa.command("npm run build", { stdio: "inherit" });

  // Run `npm pack`
  const { stdout: artifactName } = await execa.command("npm pack", {
    stdio: "pipe", // `pipe` because we want to get the name of the tarball generated by npm pack
  });

  await execa.command("ls -la", { stdio: "inherit" });

  // Build the "base" docker container that contains our CLI
  await execa.command(
    `docker build -t frontity-cli --build-arg ARTIFACT_NAME=${artifactName} .`,
    {
      stdio: "inherit",
    }
  );

  await execa.command(`rm ${artifactName}`);
});

test.concurrent(
  "in container without git",
  testContainer(async ({ runCommand }) => {
    await runCommand(
      `node_modules/.bin/frontity create --no-prompt --theme @frontity/mars-theme test-frontity-app`
    );

    let output = await runCommand("ls -a test-frontity-app");
    expect(output).toMatchInlineSnapshot(`
        ".
        ..
        README.md
        favicon.ico
        frontity.settings.js
        node_modules
        package-lock.json
        package.json
        packages"
      `);

    output = await runCommand("tree test-frontity-app/packages/");
    expect(output).toMatchSnapshot();
  })
);

test.concurrent(
  "in container with git installed but git settings are missing",
  testContainer(async ({ runCommand }) => {
    await runCommand("apk add git");
    await runCommand(
      `node_modules/.bin/frontity create --no-prompt --theme @frontity/mars-theme test-frontity-app`,
      { stdio: "inherit" }
    );

    let output = await runCommand("ls -a test-frontity-app");

    // .git is missing because user needs to set author and email in git config
    expect(output).toMatchInlineSnapshot(`
      ".
      ..
      README.md
      favicon.ico
      frontity.settings.js
      node_modules
      package-lock.json
      package.json
      packages"
    `);

    output = await runCommand("tree test-frontity-app/packages/");
    expect(output).toMatchSnapshot();
  })
);

test.concurrent(
  "in a container with git installed and configured",
  testContainer(async ({ runCommand }) => {
    await runCommand("apk add git");
    await runCommand('git config --global user.email "user@frontity.com"');
    await runCommand('git config --global user.name "Test User"');

    await runCommand(
      `node_modules/.bin/frontity create --no-prompt --theme @frontity/mars-theme test-frontity-app`,
      { stdio: "inherit" }
    );

    let output = await runCommand("ls -a test-frontity-app");
    expect(output).toMatchInlineSnapshot(`
      ".
      ..
      .git
      .gitignore
      README.md
      favicon.ico
      frontity.settings.js
      node_modules
      package-lock.json
      package.json
      packages"
    `);

    output = await runCommand("tree test-frontity-app/packages/");
    expect(output).toMatchSnapshot();
  })
);

test.concurrent(
  "in a container with git installed and configured & when a git repo already exists",
  testContainer(async ({ runCommand }) => {
    await runCommand("apk add git");
    await runCommand('git config --global user.email "user@frontity.com"');
    await runCommand('git config --global user.name "Test User"');
    await runCommand("git init test-frontity-app");

    await runCommand(
      `node_modules/.bin/frontity create --no-prompt --theme @frontity/mars-theme test-frontity-app`,
      { stdio: "inherit" }
    );

    let output = await runCommand("ls -a test-frontity-app");
    expect(output).toMatchInlineSnapshot(`
      ".
      ..
      .git
      .gitignore
      README.md
      favicon.ico
      frontity.settings.js
      node_modules
      package-lock.json
      package.json
      packages"
    `);

    // The .gitignore should be the same as the template file
    const gitignore = await runCommand("cat test-frontity-app/.gitignore");
    expect(gitignore).toEqual(
      await readFile(
        resolvePath(__dirname, "../templates/gitignore-template"),
        {
          encoding: "utf8",
        }
      )
    );

    output = await runCommand("tree test-frontity-app/packages/");
    expect(output).toMatchSnapshot();
  })
);

test.concurrent(
  "in a container with git installed and configured & when a git repo and .gitignore already exist",
  testContainer(async ({ runCommand }) => {
    await runCommand("apk add git");
    await runCommand('git config --global user.email "user@frontity.com"');
    await runCommand('git config --global user.name "Test User"');
    await runCommand("git init test-frontity-app");
    await runCommand('echo "test" > test-frontity-app/.gitignore');

    await runCommand(
      `node_modules/.bin/frontity create --no-prompt --theme @frontity/mars-theme test-frontity-app`,
      { stdio: "inherit" }
    );

    let output = await runCommand("ls -a test-frontity-app");
    expect(output).toMatchInlineSnapshot(`
      ".
      ..
      .git
      .gitignore
      README.md
      favicon.ico
      frontity.settings.js
      node_modules
      package-lock.json
      package.json
      packages"
    `);

    output = await runCommand("cat test-frontity-app/.gitignore");

    // The first line should be `test`
    expect(output).toMatchInlineSnapshot(`
      "test
      node_modules
      build"
    `);

    output = await runCommand("tree test-frontity-app/packages/");
    expect(output).toMatchSnapshot();
  })
);
