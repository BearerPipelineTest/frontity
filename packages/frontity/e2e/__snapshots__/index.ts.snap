// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 2`] = `
"test-frontity-app/packages/
└── mars-theme
    ├── CHANGELOG.md
    ├── README.md
    ├── package.json
    ├── src
    │   ├── components
    │   │   ├── featured-media.js
    │   │   ├── header.js
    │   │   ├── index.js
    │   │   ├── link.js
    │   │   ├── list
    │   │   │   ├── index.js
    │   │   │   ├── list-item.js
    │   │   │   ├── list.js
    │   │   │   └── pagination.js
    │   │   ├── loading.js
    │   │   ├── menu-icon.js
    │   │   ├── menu-modal.js
    │   │   ├── menu.js
    │   │   ├── nav.js
    │   │   ├── page-error.js
    │   │   ├── post.js
    │   │   └── title.js
    │   └── index.js
    └── types.ts

4 directories, 21 files"
`;

exports[` 4`] = `
"test-frontity-app/packages/
└── mars-theme
    ├── CHANGELOG.md
    ├── README.md
    ├── package.json
    ├── src
    │   ├── components
    │   │   ├── featured-media.js
    │   │   ├── header.js
    │   │   ├── index.js
    │   │   ├── link.js
    │   │   ├── list
    │   │   │   ├── index.js
    │   │   │   ├── list-item.js
    │   │   │   ├── list.js
    │   │   │   └── pagination.js
    │   │   ├── loading.js
    │   │   ├── menu-icon.js
    │   │   ├── menu-modal.js
    │   │   ├── menu.js
    │   │   ├── nav.js
    │   │   ├── page-error.js
    │   │   ├── post.js
    │   │   └── title.js
    │   └── index.js
    └── types.ts

4 directories, 21 files"
`;

exports[` 5`] = `
"test-frontity-app/packages/
└── mars-theme
    ├── CHANGELOG.md
    ├── README.md
    ├── package.json
    ├── src
    │   ├── components
    │   │   ├── featured-media.js
    │   │   ├── header.js
    │   │   ├── index.js
    │   │   ├── link.js
    │   │   ├── list
    │   │   │   ├── index.js
    │   │   │   ├── list-item.js
    │   │   │   ├── list.js
    │   │   │   └── pagination.js
    │   │   ├── loading.js
    │   │   ├── menu-icon.js
    │   │   ├── menu-modal.js
    │   │   ├── menu.js
    │   │   ├── nav.js
    │   │   ├── page-error.js
    │   │   ├── post.js
    │   │   └── title.js
    │   └── index.js
    └── types.ts

4 directories, 21 files"
`;

exports[` 6`] = `
"test-frontity-app/packages/
└── mars-theme
    ├── CHANGELOG.md
    ├── README.md
    ├── package.json
    ├── src
    │   ├── components
    │   │   ├── featured-media.js
    │   │   ├── header.js
    │   │   ├── index.js
    │   │   ├── link.js
    │   │   ├── list
    │   │   │   ├── index.js
    │   │   │   ├── list-item.js
    │   │   │   ├── list.js
    │   │   │   └── pagination.js
    │   │   ├── loading.js
    │   │   ├── menu-icon.js
    │   │   ├── menu-modal.js
    │   │   ├── menu.js
    │   │   ├── nav.js
    │   │   ├── page-error.js
    │   │   ├── post.js
    │   │   └── title.js
    │   └── index.js
    └── types.ts

4 directories, 21 files"
`;

exports[` 7`] = `
"test-frontity-app/packages/
└── mars-theme
    ├── CHANGELOG.md
    ├── README.md
    ├── package.json
    ├── src
    │   ├── components
    │   │   ├── featured-media.js
    │   │   ├── header.js
    │   │   ├── index.js
    │   │   ├── link.js
    │   │   ├── list
    │   │   │   ├── index.js
    │   │   │   ├── list-item.js
    │   │   │   ├── list.js
    │   │   │   └── pagination.js
    │   │   ├── loading.js
    │   │   ├── menu-icon.js
    │   │   ├── menu-modal.js
    │   │   ├── menu.js
    │   │   ├── nav.js
    │   │   ├── page-error.js
    │   │   ├── post.js
    │   │   └── title.js
    │   └── index.js
    └── types.ts

4 directories, 21 files"
`;

exports[` 9`] = `
"test-frontity-app/packages/
└── mars-theme
    ├── CHANGELOG.md
    ├── README.md
    ├── package.json
    ├── src
    │   ├── components
    │   │   ├── featured-media.js
    │   │   ├── header.js
    │   │   ├── index.js
    │   │   ├── link.js
    │   │   ├── list
    │   │   │   ├── index.js
    │   │   │   ├── list-item.js
    │   │   │   ├── list.js
    │   │   │   └── pagination.js
    │   │   ├── loading.js
    │   │   ├── menu-icon.js
    │   │   ├── menu-modal.js
    │   │   ├── menu.js
    │   │   ├── nav.js
    │   │   ├── page-error.js
    │   │   ├── post.js
    │   │   └── title.js
    │   └── index.js
    └── types.ts

4 directories, 21 files"
`;

exports[` 11`] = `
"test-frontity-app/packages/
└── mars-theme
    ├── CHANGELOG.md
    ├── README.md
    ├── package.json
    ├── src
    │   ├── components
    │   │   ├── featured-media.js
    │   │   ├── header.js
    │   │   ├── index.js
    │   │   ├── link.js
    │   │   ├── list
    │   │   │   ├── index.js
    │   │   │   ├── list-item.js
    │   │   │   ├── list.js
    │   │   │   └── pagination.js
    │   │   ├── loading.js
    │   │   ├── menu-icon.js
    │   │   ├── menu-modal.js
    │   │   ├── menu.js
    │   │   ├── nav.js
    │   │   ├── page-error.js
    │   │   ├── post.js
    │   │   └── title.js
    │   └── index.js
    └── types.ts

4 directories, 21 files"
`;

exports[`in a container with git installed and configured & when a git repo already exists 2`] = `
"test-frontity-app/packages/
└── mars-theme
    ├── CHANGELOG.md
    ├── README.md
    ├── package.json
    ├── src
    │   ├── components
    │   │   ├── featured-media.js
    │   │   ├── header.js
    │   │   ├── index.js
    │   │   ├── link.js
    │   │   ├── list
    │   │   │   ├── index.js
    │   │   │   ├── list-item.js
    │   │   │   ├── list.js
    │   │   │   └── pagination.js
    │   │   ├── loading.js
    │   │   ├── menu-icon.js
    │   │   ├── menu-modal.js
    │   │   ├── menu.js
    │   │   ├── nav.js
    │   │   ├── page-error.js
    │   │   ├── post.js
    │   │   └── title.js
    │   └── index.js
    └── types.ts

4 directories, 21 files"
`;

exports[`in a container with git installed and configured & when a git repo already exists 5`] = `
"test-frontity-app/packages/
└── mars-theme
    ├── CHANGELOG.md
    ├── README.md
    ├── package.json
    ├── src
    │   ├── components
    │   │   ├── featured-media.js
    │   │   ├── header.js
    │   │   ├── index.js
    │   │   ├── link.js
    │   │   ├── list
    │   │   │   ├── index.js
    │   │   │   ├── list-item.js
    │   │   │   ├── list.js
    │   │   │   └── pagination.js
    │   │   ├── loading.js
    │   │   ├── menu-icon.js
    │   │   ├── menu-modal.js
    │   │   ├── menu.js
    │   │   ├── nav.js
    │   │   ├── page-error.js
    │   │   ├── post.js
    │   │   └── title.js
    │   └── index.js
    └── types.ts

4 directories, 21 files"
`;

exports[`in container without git 2`] = `
"test-frontity-app/packages/
└── mars-theme
    ├── CHANGELOG.md
    ├── README.md
    ├── package.json
    ├── src
    │   ├── components
    │   │   ├── featured-media.js
    │   │   ├── header.js
    │   │   ├── index.js
    │   │   ├── link.js
    │   │   ├── list
    │   │   │   ├── index.js
    │   │   │   ├── list-item.js
    │   │   │   ├── list.js
    │   │   │   └── pagination.js
    │   │   ├── loading.js
    │   │   ├── menu-icon.js
    │   │   ├── menu-modal.js
    │   │   ├── menu.js
    │   │   ├── nav.js
    │   │   ├── page-error.js
    │   │   ├── post.js
    │   │   └── title.js
    │   └── index.js
    └── types.ts

4 directories, 21 files"
`;

exports[`in container without git 4`] = `
"test-frontity-app/packages/
└── mars-theme
    ├── CHANGELOG.md
    ├── README.md
    ├── package.json
    ├── src
    │   ├── components
    │   │   ├── featured-media.js
    │   │   ├── header.js
    │   │   ├── index.js
    │   │   ├── link.js
    │   │   ├── list
    │   │   │   ├── index.js
    │   │   │   ├── list-item.js
    │   │   │   ├── list.js
    │   │   │   └── pagination.js
    │   │   ├── loading.js
    │   │   ├── menu-icon.js
    │   │   ├── menu-modal.js
    │   │   ├── menu.js
    │   │   ├── nav.js
    │   │   ├── page-error.js
    │   │   ├── post.js
    │   │   └── title.js
    │   └── index.js
    └── types.ts

4 directories, 21 files"
`;

exports[`in container without git 6`] = `
"test-frontity-app/packages/
└── mars-theme
    ├── CHANGELOG.md
    ├── README.md
    ├── package.json
    ├── src
    │   ├── components
    │   │   ├── featured-media.js
    │   │   ├── header.js
    │   │   ├── index.js
    │   │   ├── link.js
    │   │   ├── list
    │   │   │   ├── index.js
    │   │   │   ├── list-item.js
    │   │   │   ├── list.js
    │   │   │   └── pagination.js
    │   │   ├── loading.js
    │   │   ├── menu-icon.js
    │   │   ├── menu-modal.js
    │   │   ├── menu.js
    │   │   ├── nav.js
    │   │   ├── page-error.js
    │   │   ├── post.js
    │   │   └── title.js
    │   └── index.js
    └── types.ts

4 directories, 21 files"
`;
